### DISCUSSION TITLE
Please install ya

### DISCUSSION THREAD

**Discussion by GregoryShields** [mikavilpas/yazi.nvim#602] [Q&A] [answered]
My computer is a Mac. I've installed the plugin and am using the sample lazy.nvim script which includes some keymaps.

When I press `<leader>-` the only thing that happens is I see a message on the Command Line that says:
Please install ya (the yazi command line utility). Check the documentation for more information

I've searched the documentation and everywhere else I can think of and and can find nothing about this. It's absolutely bewildering to me how other people are finding this and yet I'm apparently not intelligent enough to do it. Help?

**Comment by mikavilpas** [comment]
Hi, it looks like you might indeed be missing `ya`, which is a small command line program that should be installed with yazi. It's like yazi's little brother, and it's included in the latest yazi mac package: 

<img width="253" alt="image" src="https://github.com/user-attachments/assets/46c576e3-0e48-4cea-9da1-2e16f920a286">

How did you install yazi? What version of it do you have?

**Reply by GregoryShields** [reply]
Hi, and thanks for responding!

It was sort of an organic process of stumbling upon Yazi and using the first installation method I found, which was the exact MacPorts command that appears on this [installation](https://yazi-rs.github.io/docs/installation) page, which is:
```bash
sudo port install yazi ffmpegthumbnailer ffmpeg 7zip jq poppler fd ripgrep fzf zoxide ImageMagick
```
Normally I use Homebrew, but somehow I thought there wasn't a Homebrew method for installing Yazi, even though there is.
I chose yes to all prompts to install dependencies, which was a lot of stuff.

When I run
```bash
yazi --version
```
I get:
Yazi 0.3.3 (VERGEN_IDEMPOTENT_OUTPUT 2024-09-07)

Yeah, I see what you mean about ya being installed automatically. I suppose I need to know how to remove this installation and get the Mac one.

**Reply by mikavilpas** [reply]
I see. I think it uses this file to provide yazi https://github.com/macports/macports-ports/blob/master/sysutils/yazi/Portfile , but I can't tell if it somehow includes `ya` or not.

@sxyazi do you happen to know if installing `ya` is available via macports? I'm not familiar with this installation method. I can raise an issue if needed, but don't really know where to do it (and if it's needed) 🙂

**Reply by sxyazi** [reply]
Sorry, I don't use MacPorts myself, but from the build script, it seems like `ya` is missing. 

I might be wrong, so it's best to reach out to the MacPorts maintainers to confirm.

cc @herbygillot

**Reply by GregoryShields** [reply]
As I mentioned, I usually use Homebrew. This was the first time I've ever used MacPorts myself. I wonder whether trying the Homebrew method of installing the most recent code would install over the top of my existing installation successfully?
```bash
brew install yazi --HEAD
```
Or is there a build-from-source method that you believe would be better? I'm just trying to figure out where to go from here, and I'm not at all married to MacPorts.

Actually, I've decided to try the `brew` method with my fingers crossed and see what happens. I'll report back here when the installation finishes.

**Reply by mikavilpas** [reply, marked as answer]
It looks like https://formulae.brew.sh/formula/yazi lists the most recent version (0.4.0, released today 🎉 ). In general, I have had good experiences with homebrew and it seems to install everything in `/opt/homebrew/bin/` so I _guess_ the chance of conflicts is low.

**Reply by GregoryShields** [reply]
It worked!

You guys are great. I admire your work.

Now to learn better how to use it. I'm especially interested in using Yazi to replace other file managers inside Neovim so that I can uninstall them. I suppose I'll discover whether that is a reasonable goal fairly soon.

**Reply by herbygillot** [reply]
> Sorry, I don't use MacPorts myself, but from the build script, it seems like `ya` is missing.
> 
> I might be wrong, so it's best to reach out to the MacPorts maintainers to confirm.
> 
> cc @herbygillot

Thanks for bringing this to my attention @sxyazi , @mikavilpas.  The MacPorts port for `yazi` has just been updated to 0.4.0 [here](https://github.com/macports/macports-ports/commit/ef38ad0079e556817c16fd246ff7293de5c7677c), and now installs `ya` as expected.

**Reply by sxyazi** [reply]
Awesome, thanks for the fix and update for 0.4 @herbygillot!

**Reply by GregoryShields** [reply]
> Thanks for bringing this to my attention @sxyazi , @mikavilpas. The MacPorts port for `yazi` has just been updated to 0.4.0 [here](https://github.com/macports/macports-ports/commit/ef38ad0079e556817c16fd246ff7293de5c7677c), and now installs `ya` as expected.

Well, now I'm wondering whether I should go back and install from MacPorts again, as the install I did from Homebrew is still at version 0.3.3. In any case, it's great to see this level of cooperation. If I weren't in the situation I'm in, I'd probably be trying to send all of you some money.

@herbygillot: Actually I did just now attempt to upgrade Yazi using the `sudo port` command, but was unable to get the 0.4.0 version. Maybe I need to use the provided link to build from source? I'm just not sure how to proceed.

**Reply by herbygillot** [reply]
@GregoryShields - you may need to update. Do `sudo port selfupdate`, then try `sudo port upgrade yazi`

**Reply by GregoryShields** [reply]
@herbygillot I had done that before, but this time it worked when I ran `sudo port upgrade outdated`. I guess the upgrade wasn't there when I did it before.
Thanks.

**Reply by herbygillot** [reply]
> @herbygillot I had done that before, but this time it worked when I ran `sudo port upgrade outdated`. I guess the upgrade wasn't there when I did it before. Thanks.

MacPorts (the `ports` command) maintains a local catalog of available software and versions. When you do `port install foo`, the source and current version of `foo` is installed from that local catalog.

Hence why the `selfupdate` command.  `selfupdate` updates both the `port` command itself and the local catalog.

**Then** when you do an `install` or `upgrade`, it's installing or upgrading the latest version defined in the freshly updated local catalog.

`upgrade outdated` updates every piece of software that you installed with MacPorts that is now out-of-date compared to your local catalog.

After a `selfupdate`, you can do `sudo port echo outdated` to see the list of what `ports` now considers outdated.
